web/src/test/java/io/fabric8/launcher/web/endpoints/osio/OsioEmptyLaunchEndpointIT.java

```diff
@@ -0,0 +1,251 @@
+import io.fabric8.launcher.service.openshift.impl.Fabric8OpenShiftServiceFactory;
+import io.fabric8.launcher.service.openshift.impl.OpenShiftClusterRegistryImpl;
+import io.fabric8.launcher.service.openshift.spi.OpenShiftServiceSpi;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.response.ResponseBodyExtractionOptions;
+import io.restassured.specification.RequestSpecification;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
+import org.hamcrest.core.IsNull;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+
+import static io.fabric8.launcher.web.endpoints.osio.OsioTests.getOsioIdentity;
+import static io.fabric8.launcher.web.endpoints.osio.OsioTests.getWitClient;
+import static io.restassured.RestAssured.given;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class OsioEmptyLaunchEndpointIT {
+
+    private static final Logger LOG = Logger.getLogger(OsioLaunchEndpointIT.class.getName());
+
+    private static final String TEST_ID = randomAlphanumeric(5).toLowerCase();
+
+    private static final String SPACE_NAME = "space-osio-it-" + TEST_ID;
+
+    private static final String LAUNCH_PROJECT_NAME = "project-osio-it-launch-empty-" + TEST_ID;
+
+    private static final String LAUNCH_MISSION = "rest-http";
+
+    private static final String LAUNCH_RUNTIME = "vert.x";
+
+    private static final String LAUNCH_RUNTIME_VERSION = "community";
+
+    private static final String IMPORT_PROJECT_NAME = "project-osio-it-import-" + TEST_ID;
+
+
+    private static final List<String> REPOSITORY_TO_CLEAN = Arrays.asList(LAUNCH_PROJECT_NAME, IMPORT_PROJECT_NAME);
+
+    private static final List<String> PROJECT_TO_CLEAN = Arrays.asList(LAUNCH_PROJECT_NAME, IMPORT_PROJECT_NAME);
+
+    @Rule
+    public final TemporaryFolder tmpFolder = new TemporaryFolder();
+
+    @ArquillianResource
+    private URI deploymentUri;
+
+    private static Space space;
+
+    @Deployment(testable = false)
+    public static Archive<?> createDeployment() {
+        return ShrinkWrap.createFromZipFile(WebArchive.class, new File("target/launcher-backend.war"));
+    }
+
+    @Test
+    public void should_launch() throws Exception {
```

Comment:
Move this test to OsioLaunchEndpointIT under a new method and make sure it tests that the repository is empty